File: app/src/main/java/com/example/assignment/DatabaseDao.java
package com.example.assignment;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface DatabaseDao {
    @Query("SELECT * FROM treasures")
    List<Treasure> getAllTreasures();

    @Insert
    void insert(Treasure treasure);
}

File: app/src/main/java/com/example/assignment/DatabaseDao.kt
/*package com.example.assignment

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

@Dao
interface DatabaseDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertLogEntry(Treasure: LogEntry)

    @Query("SELECT * FROM Treasure")
    fun getAll(): LiveData<List<Treasure>>

    @Query("DELETE FROM Treasure WHERE Treasure = :selected")
    fun deleteMovies(watched: Boolean)

    @Update
    fun updateMovie(Treasure: Treasure)

    @Query("DELETE FROM Treasure WHERE id = :id")
    fun delete(id: Int?)
}*/